modal-js/src/file-handle.ts

```diff
@@ -0,0 +1,155 @@
+import { SeekWhence } from "../proto/modal_proto/api";
+import { client } from "./client";
+
+/** File open modes supported by the filesystem API. */
+export type FileMode = "r" | "w" | "a" | "r+" | "w+" | "a+";
+
+/** Options for file operations. */
+export type FileOptions = {
```

Comment:
Probably want one type for each Options needed on a method

Reply:
Do you mean like `ReadOptions` / `WriteOptions`?